nsv13/code/modules/overmap/knpc.dm

```diff
@@ -339,12 +352,13 @@ This is to account for sec Ju-Jitsuing boarding commandos.
 		var/dist = get_dist(O, H)
 		if(istype(O, /obj/structure/closet) && dist <= best_distance)
 			var/obj/structure/closet/C = O
-			target_item = locate(/obj/item/gun) in C.contents
-			if(target_item && C.allowed(H))
+			var/obj/item/gun/G = locate(/obj/item/gun) in C.contents
+			if(C.allowed(H) && gun_suitable(H, G))
+				target_item = G
```

Comment:
It's better to null check here instead of in gun\_suitable(). Callers should handle null args, not procs